src/cljs/athens/views/block_page.cljs

```diff
@@ -68,15 +72,29 @@
 
 ;;; Components
 
+(defn handle-enter
+  [_ uid _]
+  (let [new-uid   (gen-block-uid)
+        now       (now-ts)]
+    (dispatch [:transact [{:block/uid       uid
+                           :edit/time       now
+                           :block/children  [{:block/order  0
+                                              :block/uid    new-uid
+                                              :block/open   true
+                                              :block/string ""}]}]])
+    (dispatch [:editing/uid new-uid])))
 
 (defn block-page-key-down
-  [_ _ _]
-  (prn "TODO: block-page-key-down"))
+  [e uid state]
+  (let [d-event (destruct-event e)
+        {:keys [key-code]} d-event]
+    (cond
+      (= key-code KeyCodes.ENTER) (handle-enter e uid state))))
 
 
 (defn block-page-change
   [e _uid state]
-  (let [value (.. e -target -value)]
+  (let [value (clojure.string/trim-newline (.. e -target -value))]
```

Comment:
I was getting new lines because that value would get recorded before the on-key-down would fire. So trimming them seems appropriate?

Reply:
I believe new lines occur as default behavior from ENTER. If you use `(.. e preventDefault`), this won't happen.

https://github.com/athensresearch/athens/blob/0579308dea4851c0e5ae9c44ddf8f2ffb9e06710/src/cljs/athens/keybindings.cljs#L270-L291

Reply:
I believe new lines occur as default behavior from ENTER. If you use `(.. e preventDefault`), this won't happen.

https://github.com/athensresearch/athens/blob/0579308dea4851c0e5ae9c44ddf8f2ffb9e06710/src/cljs/athens/keybindings.cljs#L270-L291